connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java

```diff
@@ -109,6 +121,69 @@ public void testThreadName() {
                 .newThread(EMPTY_RUNNABLE).getName().startsWith(FileOffsetBackingStore.class.getSimpleName()));
     }
 
+    @Test
+    public void testConnectorPartitions() throws Exception {
+        @SuppressWarnings("unchecked")
+        Callback<Void> setCallback = mock(Callback.class);
+
+        JsonConverter jsonConverter = new JsonConverter();
+        jsonConverter.configure(Collections.singletonMap(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, "false"), true);
+
+        Map<ByteBuffer, ByteBuffer> serializedPartitionOffsets = new HashMap<>();
+        serializedPartitionOffsets.put(
+                ByteBuffer.wrap(jsonConverter.fromConnectData("", null,
+                        Arrays.asList("connector1", Collections.singletonMap("partitionKey", "partitionValue1")))),
+                ByteBuffer.wrap(jsonConverter.fromConnectData("", null,
+                        Collections.singletonMap("offsetKey", "offsetValue")))
```

Comment:
This would be a bit easier to read with a couple small utility methods:

```java
private static ByteBuffer serializeKey(Converter converter, String connectorName, Map<String, Object> sourcePartition) {
    List<Object> nameAndPartition = Arrays.asList(connectorName, sourcePartition);
    return serialize(converter, nameAndPartition);
}

private static ByteBuffer serialize(Converter converter, Object value) {
    byte[] serialized = converter.fromConnectData("", null, value);
    return ByteBuffer.wrap(serialized);
}
```

We could then turn lines like this into:
```suggestion
                serializeKey(jsonConverter, "connector1", Collections.singletonMap("partitionKey", "partitionValue1")),
                serialize(jsonConverter, Collections.singletonMap("offsetKey", "offsetValue"))
```

Reply:
Nice 👍